### PR TITLE
test(ARCO-202): Unit test for callbacker sender

### PR DESCRIPTION
**Pull request by kameshraj23**
## Description of Changes

Adding unit test for callerback sender

### CODE REVIEW

File: internal/callbacker/sender_test.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+		t.Run(tc.name, func(t *testing.T) {
+			// Given
+			retries := 0
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+				w.WriteHeader(tc.responseStatus)
+				retries++
+			}))
+			defer server.Close()
+			logger := slog.New(slog.NewTextHandler(new(bytes.Buffer), nil))
+			sender, err := callbacker.NewSender(server.Client(), logger)
```
Comment (boecklim) [inline internal/callbacker/sender_test.go:41]: Rename `sender` to `sut`

File: internal/callbacker/sender_test.go [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+		responseStatus  int
+		expectedSuccess bool
+		expectedRetries int
+	}{
+		{
+			name:            "success - callback sent",
+			responseStatus:  http.StatusOK,
+			expectedSuccess: true,
+			expectedRetries: 1,
+		},
```
Comment (boecklim) [inline internal/callbacker/sender_test.go:28]: Add another case where `responseStatus` is an error e.g. `http.StatusInternalServerError`

File: internal/callbacker/sender_test.go [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+
+	// Then: Assert that sender is disposed
+	err = sender.Health()
+	require.EqualError(t, err, "callbacker is disposed already")
+}
+
+func TestCallbackSender_Health(t *testing.T) {
+	// Given
+	logger := slog.New(slog.NewTextHandler(new(bytes.Buffer), nil))
+	sender, err := callbacker.NewSender(http.DefaultClient, logger)
```
Comment (boecklim) [inline internal/callbacker/sender_test.go:73]: Rename `sender` to `sut`

File: internal/callbacker/sender_test.go [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+			require.Equal(t, tc.expectedSuccess, success)
+		})
+	}
+}
+
+func TestCallbackSender_Send_WithRetries(t *testing.T) {
+	// Given
+	client := &http.Client{}
+	logger := slog.New(slog.NewJSONHandler(bytes.NewBuffer([]byte{}), nil))
+	sender, _ := callbacker.NewSender(client, logger)
```
Comment (boecklim) [inline internal/callbacker/sender_test.go:142]: Rename `sender` to `sut`

File: internal/callbacker/sender_test.go [hunk 5]
```diff
@@ -0,0 +1,161 @@
...
+				err := json.NewDecoder(r.Body).Decode(&batch)
+				require.NoError(t, err)
+				require.Equal(t, len(tc.dtos), batch.Count)
+
+				w.WriteHeader(tc.responseStatus)
+			}))
+			defer server.Close()
+
+			logger := slog.New(slog.NewTextHandler(new(bytes.Buffer), nil))
+			sender, err := callbacker.NewSender(server.Client(), logger)
```
Comment (boecklim) [inline internal/callbacker/sender_test.go:125]: Rename `sender` to `sut`

File: internal/callbacker/sender_test.go [hunk 6]
```diff
@@ -0,0 +1,161 @@
...
+		expectedSuccess bool
+	}{
+		{
+			name:            "success - batch sent",
+			url:             "/batch",
+			token:           "test-token",
+			dtos:            []*callbacker.Callback{{TxID: "1234", TxStatus: "SEEN_ON_NETWORK"}, {TxID: "5678", TxStatus: "MINED"}},
+			responseStatus:  http.StatusOK,
+			expectedSuccess: true,
+		},
```
Comment (boecklim) [inline internal/callbacker/sender_test.go:106]: Add another case where `responseStatus` is an error e.g. `http.StatusInternalServerError`

File: internal/callbacker/sender_test.go [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+			success := sut.Send(server.URL, "test-token", &callbacker.Callback{TxID: "1234", TxStatus: "SEEN_ON_NETWORK"})
+			//Then
+			require.Equal(t, tc.expectedSuccess, success, "Expected success to be %v, but got %v", tc.expectedSuccess, success)
+			require.Equal(t, tc.expectedRetries, retries, "Expected retries to be %d, but got %d", tc.expectedRetries, retries)
+		})
+	}
+}
+func TestCallbackSender_GracefulStop(t *testing.T) {
+	// Given
+	logger := slog.New(slog.NewTextHandler(new(bytes.Buffer), nil))
```
Comment (arkadiuszos4chain) [inline internal/callbacker/sender_test.go:71]: You can use `slog.Default()`

**Review by boecklim — DISMISSED** [review]

**Review by boecklim — APPROVED** [review]

**Pull request merged**